pyacq/core/nodegroup.py

```diff
@@ -1,93 +1,161 @@
 from pyqtgraph.Qt import QtCore, QtGui
+from pyqtgraph.util.mutex import Mutex
 
 from .rpc import RPCServer
-from .node import Node, WidgetNode
+from .node import Node, WidgetNode, register_node_type
 from .nodelist import all_nodes
 
+import time
+import importlib
+import pickle
+import weakref
+import random
+import string
+import zmq
 
-
-class RpcThread( QtCore.QThread):
-    def __init__(self, rpc_server, parent = None):
+class RpcThreadSocket( QtCore.QThread):
+    new_message = QtCore.Signal(QtCore.QByteArray, QtCore.QByteArray)
+    def __init__(self, rpc_server, local_addr, parent = None):
         QtCore.QThread.__init__(self, parent)
-        self.rpc_server = rpc_server
+        self.rpc_socket = weakref.ref(rpc_server._socket)
+        
+        context = zmq.Context.instance()
+        self.local_socket = context.socket(zmq.PAIR)
```

Comment:
This class looks great!
One minor issue: the sockets and poller should be created in the thread where they are used.